test/examples.js

```diff
@@ -0,0 +1,487 @@
+'use strict';
+
+const metasync = require('..');
+const fs = require('fs');
+const metatests = require('metatests');
+const ASYNC_TIMEOUT = 200;
+
+// Data Collector
+
+metatests.test('dataCollector / simple', test => {
+
+  const dataCollector = metasync
+    .collect(4)
+    .done((err, data) => {
+      test.error(err);
+      test.strictSame(Object.keys(data).length, 4);
+      test.end();
+    });
+
+  dataCollector.collect('user', null, { name: 'Marcus Aurelius' });
+
+  fs.readFile('README.md', test.mustCall((err, data) => {
+    dataCollector.collect('readme', err, data);
+  }));
+
+  fs.readFile('AUTHORS', test.mustCall((err, data) => {
+    dataCollector.collect('authors', err, data);
+  }));
+
+  setTimeout(test.mustCall(() => {
+    dataCollector.pick('timer', { date: new Date() });
+  }), ASYNC_TIMEOUT);
+});
+
+metatests.test('data collector / timeout', test => {
+
+  const expectedErr = new Error('Metasync: Collector timed out');
+  const expectedData = { user: { name: 'Marcus Aurelius' } };
+
+  const dataCollector = metasync
+    .collect(4)
+    .timeout(1000)
+    .done((err, data) => {
+      test.isError(err, expectedErr);
+      test.strictSame(data, expectedData);
+      test.end();
+    });
+
+  dataCollector.pick('user', { name: 'Marcus Aurelius' });
+
+});
+
+metatests.test('data collector / error', test => {
+
+  const expectedErr = new Error('User not found');
+  const expectedData = { file: 'file content' };
+
+  const dataCollector = metasync
+    .collect(4)
+    .done((err, data) => {
+      test.isError(err, expectedErr);
+      test.strictSame(data, expectedData);
+      test.end();
+    });
+
+  dataCollector.pick('file', 'file content');
+  dataCollector.fail('user', new Error('User not found'));
+  dataCollector.pick('score', 1000);
+  dataCollector.fail('tcp', new Error('No socket'));
+
+});
+
+// Key Collector
+
+metatests.test('key collector / simple', test => {
+
+  const keyCollector = metasync
+    .collect(['user', 'readme'])
+    .timeout(1000)
+    .done((err, data) => {
+      test.error(err);
+      test.strictSame(Object.keys(data).length, 2);
+      test.end();
+    });
+
+  keyCollector.pick('user', { name: 'Marcus Aurelius' });
+
+  fs.readFile('README.md', (err, data) => {
+    test.error(err);
+    keyCollector.pick('readme', data);
+  });
+
+});
+
+// Parallel execution
+
+metatests.test('parallel', test => {
+
+  test.plan(5);
```

Comment:
Can be replaced with mustCall I think.

Reply:
Now we can't do so, because ``metasync.paralell`` use the property ``fn.length`` for each functions in the composition. It now have ``fn.length === 2``. But if we wrap each functions in ``test.mustCall``, it'll be have ``fn.length === 0``, because ``mustCall`` use rest operator for arhuments. Also, as for me, it may have problems later, like this